File: apps/mobile-portfolio/ios/mobileportfolio/NativeSparkline/SparklineTheme.swift
import UIKit

enum SparklineTheme {

    static func lineColor(for trend: String) -> UIColor {
        switch trend {
        case "up": return UIColor(red: 255/255, green: 59/255, blue: 48/255, alpha: 1)
        case "down": return UIColor(red: 52/255, green: 199/255, blue: 89/255, alpha: 1)
        default: return UIColor(red: 142/255, green: 142/255, blue: 147/255, alpha: 1)
        }
    }

    static func baselineColor(for trend: String) -> UIColor {
        let lineColor = lineColor(for: trend)
        return lineColor.withAlphaComponent(0.3)
    }

    static func gradientColors(for trend: String) -> [UIColor] {
        let lineColor = lineColor(for: trend)
        return [
            lineColor.withAlphaComponent(0.15),
            lineColor.withAlphaComponent(0.0)
        ]
    }

    static var baselineDashLengths: [CGFloat] { [3, 3] }
}

File: apps/mobile-portfolio/android/app/src/main/java/com/anonymous/mobileportfolio/view/sparkline/SparklineTheme.kt
package com.anonymous.mobileportfolio.view.sparkline

import android.graphics.Color

object SparklineTheme {

    private const val UP_COLOR = "#ff3b30"
    private const val DOWN_COLOR = "#34c759"
    private const val FLAT_COLOR = "#8e8e93"

    fun lineColorForTrend(trend: String): Int = when (trend) {
        "up" -> Color.parseColor(UP_COLOR)
        "down" -> Color.parseColor(DOWN_COLOR)
        else -> Color.parseColor(FLAT_COLOR)
    }

    fun baselineColor(trend: String): Int {
        val baseColor = lineColorForTrend(trend)
        return Color.argb(
            (Color.alpha(baseColor) * 0.3).toInt(),
            Color.red(baseColor),
            Color.green(baseColor),
            Color.blue(baseColor)
        )
    }

    fun gradientStartColor(trend: String): Int {
        val baseColor = lineColorForTrend(trend)
        return Color.argb(
            (Color.alpha(baseColor) * 0.15).toInt(),
            Color.red(baseColor),
            Color.green(baseColor),
            Color.blue(baseColor)
        )
    }

    fun gradientEndColor(trend: String): Int {
        val baseColor = lineColorForTrend(trend)
        return Color.argb(
            0,
            Color.red(baseColor),
            Color.green(baseColor),
            Color.blue(baseColor)
        )
    }
}
